london-mayors-questions/questions.py

```diff
@@ -0,0 +1,816 @@
+
+    # This loops through each membership, checks to see if it's for the Assembly, if so adds it to the map
+
+    person_ids_by_name = {}
+
+    for membership in members_raw_data['memberships']:
+        if membership.get('organization_id') == 'london-assembly':
+            name = getNameFromPerson(people_by_id[membership['person_id']])
+
+            if name not in person_ids_by_name:
+                person_ids_by_name[name] = membership['person_id']
+                logger.debug(u'Added ID map for for {}'.format(name))
+            else:
+                if person_ids_by_name[name] != membership['person_id']:
+                    raise Exception('Multiple people with name {}'.format(name))
+
+    logger.debug('Added {} names with Assembly memberships'.format(len(person_ids_by_name)))
+
+    return person_ids_by_name
+
+
+def getNameFromPerson(person):
+
+    for name in person.get('other_names', []):
+        if name['note'] == 'Main':
+            return name['given_name'] + ' ' + name['family_name']
+
+    raise Exception('Unable to find main name for person {}'.format(person['id']))
+
+
+@click.group()
+@click_log.simple_verbosity_option(logger, default='warning')
+def cli():
+    pass
+
+
+@cli.command()
+@click.option('-s', '--start', type=CLI_DATETIME_FORMAT, help='The first date of the range to be scrape.')
+@click.option('-e', '--end', type=CLI_DATETIME_FORMAT, help='The last date of the range to be scraped.')
+@click.option('--force-scrape-dates', is_flag=True, help='Force all dates in the range to be re-scraped regardless of status')
+@click.option('--force-refresh-questions', is_flag=True, help='Force all detected questions to have their state refreshed')
+def meetings(start, end, force_scrape_dates, force_refresh_questions):
+    ''' Get a list of questions from the London Assembly website asked between the dates given. '''
+
+    logger.info('Scraping London Assembly')
+
+    if start:
+        start_date = start.date()
+        logger.debug('End date has been explicitly set to {} by CLI'.format(start_date))
+    else:
+        start_date = datetime.datetime.strptime(DEFAULT_START_DATE, '%Y-%m-%d').date()
+        logger.debug('Start date has been automatically set to {} by config'.format(start_date))
+
+    if end:
+        end_date = end.date()
+        logger.debug('End date has been explicitly set to {} by CLI'.format(end_date))
+    else:
+        # Yesterday
+        end_date = (datetime.datetime.today() - datetime.timedelta(days=1)).date()
+        logger.debug('End date has been automatically set to {} (yesterday)'.format(end_date))
+
+    if end_date < start_date:
+        logger.error('End date is before the start date. Aborting.')
+        return
+
+    dates_in_range = getDatesInRange(start_date, end_date)
+
+    logger.info('Targetting {} dates between {} and {}.'.format(len(dates_in_range), start_date, end_date))
+
+    questions_in_range = []
+
+    with click.progressbar(dates_in_range) as bar:
```

Comment:
According to [the docs](https://click.palletsprojects.com/en/7.x/api/#click.progressbar) it will only render if the output is a terminal, so I believe this will behave as expected and output nothing whilst running from a crontab.